test/global_uid_test.rb

```diff
@@ -205,6 +209,33 @@ def table_exists?(connection, table)
     end
   end
 
+  describe "Performance" do
+    before do
+      CreateWithNoParams.up
+      CreateWithoutGlobalUIDs.up
+    end
+
+    it "has a negligible performance impact" do
+      report = Benchmark.ips do |x|
+        x.report(WithGlobalUID)    { WithGlobalUID.create! }
+        x.report(WithoutGlobalUID) { WithoutGlobalUID.create! }
+        x.compare!
+      end
+
+      with_global_uid    = report.entries.find { |e| e.label == WithGlobalUID }.stats
+      without_global_uid = report.entries.find { |e| e.label == WithoutGlobalUID }.stats
+
+      performance_impact, _ = with_global_uid.slowdown(without_global_uid)
+      assert performance_impact < 2
```

Comment:
Does this mean we would accept a factor 2 performance difference? I think we can go lower – maybe 1.5?

Reply:
There's a lot of variance between each run, I've seen it perform between 1.1x and 1.8x slower with it usually averaging around 1.3x. I put 2 here so the build doesn't break and anything higher than that is a cause for concern.

I'll see if I can ratchet the value down without adding noise to CI. That was my original hope when using benchmark-ips over benchmark